fuzz/fuzz_targets/object_cache.rs

```diff
@@ -43,19 +43,47 @@ fn compute_serialized_len(a: &Allocator, n: NodePtr) -> u64 {
     *stack.last().expect("internal error, empty stack")
 }
 
+fn pick_node(a: &Allocator, root: NodePtr, mut node_idx: i32) -> NodePtr {
+    let mut stack = vec![root];
+    let mut seen_node = HashSet::<NodePtr>::new();
+
+    while let Some(node) = stack.pop() {
+        if node_idx == 0 {
+            return node;
+        }
+        if !seen_node.insert(node) {
+            continue;
+        }
+        node_idx -= 1;
+        if let SExp::Pair(left, right) = a.sexp(node) {
+            stack.push(left);
+            stack.push(right);
+        }
+    }
+    NodePtr::NIL
+}
+
 fuzz_target!(|data: &[u8]| {
     let mut unstructured = arbitrary::Unstructured::new(data);
     let mut allocator = Allocator::new();
-    let program = make_tree::make_tree(&mut allocator, &mut unstructured);
+    let (tree, node_count) =
+        make_tree::make_tree_limits(&mut allocator, &mut unstructured, 10_000, true);
 
     let mut hash_cache = ObjectCache::new(treehash);
     let mut length_cache = ObjectCache::new(serialized_length);
-    visit_tree(&allocator, program, |a, node| {
```

Comment:
this fuzzer was changed to testing a single node, picked based on the input data, rather than looping over all of them. testing all nodes risks taking a long time, timing out the fuzzer.